### PR TITLE
Resync mine orientations in JIP. 

### LINKED ISSUE #3317: Mines/Explosives orientation resets when JIP or reconnect
ACE3 Version: 3.4.2

**Mods:**
* `@CBA_A3`
* `@ace3`

**Placed ACE3 Modules:**
* None

**Description:**
* Placed mines/explosives orientation resets to zero when joining as JIP or reconnecting

**Steps to reproduce:**
* Place mine/explosive -> JIP or disconnect and reconnect back -> mine orientation reset

**Where did the issue occur?**
* Dedicated server

**RPT log file:**
* -

### PR DESCRIPTION
**Pull request by esteldunedain**
### When merged this pull request will:

1. Resync mine orientations in JIP. 
2. Fix #3317

### CODE REVIEW

File: addons/explosives/XEH_postInit.sqf [hunk 1]
```diff
@@ -36,16 +36,31 @@ GVAR(Setup) = objNull;
...
+        params ["_explosivesOrientations"];
+        TRACE_1("serverSendsOrientations received:",_explosivesOrientations);
+        {
+            params ["_explosive","_direction","_pitch"];
+            [_explosive, _direction, _pitch] call FUNC(setPosition);
+        } forEach _explosivesOrientations;
+    }] call EFUNC(common,addEventHandler);
+
+    //  Create a logic to get the client ID
+    private _logic = createGroup sideLogic createUnit ["Logic", [0,0,0], [], 0, "NONE"];
```
Comment (esteldunedain) [inline addons/explosives/XEH_postInit.sqf]: I'm still in the market for a better way of doing this. If we find it we should probably use it in ACE_Markers too

File: addons/markers/XEH_postInit.sqf [hunk 2]
```diff
@@ -9,9 +9,9 @@
 
 // request marker data for JIP
 if (isMultiplayer && {!isServer} && {hasInterface}) then {
-    private _logic = createGroup sideLogic createUnit ["Logic", [0,0,0], [], 0, "NONE"];
-
-    [QGVAR(sendMarkersJIP), [_logic]] call EFUNC(common,serverEvent);
+    GVAR(localLogic) = sideLogic createUnit ["Logic", [0,0,0], [], 0, "NONE"];
+    GVAR(localGroup) = createGroup GVAR(localLogic);
```
Comment (commy2) [inline addons/markers/XEH_postInit.sqf]: This is still wrong. I think this is what you want:

```
GVAR(localGroup) = createGroup sideLogic;
GVAR(localLogic) = GVAR(localGroup) createUnit ["Logic", [0,0,0], [], 0, "NONE"];
```

File: addons/markers/functions/fnc_setMarkerJIP.sqf [hunk 3]
```diff
@@ -51,4 +51,7 @@ TRACE_3("params",_allMapMarkers,_allMapMarkersProperties,_logic);
     false
 } count allMapMarkers;
 
-deleteVehicle _logic;
+deleteGroup GVAR(localGroup);
+GVAR(localGroup) = nil;
+deleteVehicle GVAR(localLogic);
+GVAR(localLogic) = nil;
```
Comment (commy2) [inline addons/markers/functions/fnc_setMarkerJIP.sqf]: see above:

```
private _group = group GVAR(localLogic);
deleteVehicle GVAR(localLogic);
deleteGroup _group;
```

File: addons/markers/XEH_postInit.sqf (hunk 2 above)
Comment (commy2) [inline addons/markers/XEH_postInit.sqf]: `sideLogic` returns a SIDE
`createGroup <SIDE>` returns a group
`<GROUP> createUnit <ARRAY>`returns an object

Comment (esteldunedain) [inline addons/markers/XEH_postInit.sqf]: Sorry, I just read the doc and you are completely right. I'll change it later today.

**Comment by esteldunedain** [comment]
@commy2, I think it's finally ready. Please takea look again when you have time.

**Comment by Glowbal** [comment]
@commy2 @esteldunedain please let me know if this PR is ready. It's still marked as needs review.

Also; merge conflict.

**Comment by commy2** [comment]
I will do this later today.

**Comment by esteldunedain** [comment]
I'm testing atm

**Comment by esteldunedain** [comment]
Yay, after much testing I finally got it working properly. Should be ready for merge now @Glowbal.

**Comment by Glowbal** [comment]
:+1: Merge if @commy2 is also happy

**Pull request merged**